app/workers/concerns/immediate_email_generator.rb

```diff
@@ -1,27 +1,20 @@
-class ImmediateEmailGenerationWorker
-  include Sidekiq::Worker
-
-  sidekiq_options queue: :email_generation_immediate
-
-  LOCK_NAME = "immediate_email_generation_worker".freeze
+module ImmediateEmailGenerator
+private
 
-  def perform
-    ensure_only_running_once do
-      SubscribersForImmediateEmailQuery.call.find_in_batches(batch_size: 5000) do |group|
-        subscription_contents = grouped_subscription_contents(group.pluck(:id))
-        update_content_change_cache(subscription_contents)
-        update_message_cache(subscription_contents)
-        create_content_change_emails(group, subscription_contents)
-        create_message_emails(group, subscription_contents)
-      end
+  def ensure_only_running_once(type, id)
```

Comment:
Do we need this anymore now that we're having a single worker for each content change/message? Sidekiq should make sure that only one worker picks up the job, so I don't think we would be in a situation where the job could run at the same time by multiple workers.

Reply:
I've done some thinking/digging and as far as I know/can tell Sidekiq will ensure at least one worker will pick up the job, but won't guarantee that only one will. It's theoretically possible that an instance of ProcessContentChangeWorker will queue up a job, fail, get retried and queue up _another_ job. My understanding is that the second job could then process SubscriptionContent which the first job is working on creating emails for but don't yet have one. I _think_ therefore, in theory, that removing it _could_ result in duplicate emails being sen